bin/fix-dbs-parentag.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+from __future__ import print_function, division
+from pprint import pprint
+from collections import defaultdict
+from dbs.apis.dbsClient import DbsApi
+
+endpoint = 'https://cmsweb.cern.ch/dbs/prod/global/DBSReader'
+dbs = DbsApi('https://cmsweb.cern.ch/dbs/prod/global/DBSReader')
+
+def listOfParentDatasets(dataset):
+    parents = dbs.listDatasetParents(dataset=dataset)
+    parentDatasets = []
+    for parent in parents:
+        parentDatasets.append(parent['parent_dataset'])
+    return parentDatasets
+
+def listBlocksWithNoParents(dataset):
+    allBlocks = dbs.listBlocks(dataset=dataset)
+    blockNames = []
+    for block in allBlocks:
+        blockNames.append(block['block_name'])
+    parentBlocks = dbs.listBlockParents(block_name=blockNames)
+
+    cblock = set()
+    for pblock in parentBlocks:
+        cblock.add(pblock['this_block_name'])
+
+    noParentBlocks = set(blockNames) - cblock
+    return noParentBlocks
+
+def listFilesWithNoParents(blockName):
+    allFiles = dbs.listFiles(block_name=blockName)
+    parentFiles = dbs.listFileParents(block_name=blockName)
```

Comment:
Doesn't it return a list of dictionaries (parent_logical_file_name, logical_file_name)? If so, then we can simply call `listFileParents` and check which files don't have parents.

Reply:
It doesn't return anything if there  is no parents.

Reply:
Shouldn't we file a bug for Yuyi/DBS then? I think it should just return an empty string (or None) for the parent key.

Reply:
I think that would be better for this use case but that is what it is now. I don't think it would be considered as a bug. Not sure any application using this has to be changed as well.